src/dialogs/connectsettingsdialog.cpp

```diff
@@ -38,3 +39,22 @@ void ConnectSettingsDialog::locateSerialPort()
         ui->serialPortCB->clear();
     }
 }
+
+void ConnectSettingsDialog::initFirmwareComboBox()
+{
+    QStringList fw;
+    fw.append("Auto-Detect");
+    fw.append("Marlin");
+    fw.append("Repetier");
+    fw.append("Sprinter");
+    fw.append("Teacup");
+    ui->firmwareCB->addItems(fw);
```

Comment:
We have already a list of supported plugins, do not need to create it.
The GCode namespace have a list and atcore too.